apps/condo/domains/ticket/schema/Ticket.js

```diff
@@ -456,6 +482,28 @@ const Ticket = new GQLListSchema('Ticket', {
                     }
                 }
             }
+
+            const newItem = { ...existingItem, ...resolvedData }
+            const resolvedStatus = await getById('TicketStatus', newItem.status)
+
+            if (newItem.deferredUntil) {
+                if (operation === 'create') {
+                    if (resolvedStatus.type !== DEFERRED_STATUS_TYPE) {
+                        return addValidationError(`${WRONG_VALUE} should not create ticket with "deferredUntil" field if status type is not ${DEFERRED_STATUS_TYPE}`)
+                    }
+                }
+
+                if (operation === 'update') {
+                    const existingStatus = await getById('TicketStatus', existingItem.status)
+                    if (originalInput.deferredUntil && existingStatus.type !== DEFERRED_STATUS_TYPE && resolvedStatus.type !== DEFERRED_STATUS_TYPE) {
```

Comment:
and what should happen is `existingStatus.type === DEFERRED_STATUS_TYPE && resolvedStatus.type !== DEFERRED_STATUS_TYPE` ?

Reply:
we shouldn't change the `deferredUntil` field unless it's for deferred tickets (before or after update)

Reply:
i added comment